pkg/dns/aws/dns.go

```diff
@@ -142,26 +142,31 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 	elbConfig := aws.NewConfig().WithRegion(region)
 	tagConfig := aws.NewConfig()
 
+	partition, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
+	if !ok {
```

Comment:
Not sure but did you check if partition would be nil if ok is false? The switch case would throw error in that case since it's a function call with nil reference.

Reply:
@rna-afk thanks for the review! Yes, I checked the code for `PartitionForRegion` in the SDK and an empty struct is returned in case of error https://github.com/aws/aws-sdk-go/blob/v1.44.6/aws/endpoints/endpoints.go#L384-L392. So it should be safe to call `ID()` on it, which will result in an empty string.